src/OpenApi/src/Extensions/JsonObjectSchemaExtensions.cs

```diff
@@ -0,0 +1,279 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.ComponentModel.DataAnnotations;
+using System.Globalization;
+using System.Linq;
+using System.Reflection;
+using System.Text.Json.Nodes;
+using JsonSchemaMapper;
+using Microsoft.AspNetCore.Mvc.ApiExplorer;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.AspNetCore.Routing.Constraints;
+using Microsoft.OpenApi.Models;
+
+namespace Microsoft.AspNetCore.OpenApi;
+
+/// <summary>
+/// Provides a set of extension methods for modifying the opaque JSON Schema type
+/// that is provided by the underlying schema generator in System.Text.Json.
+/// </summary>
+internal static class JsonObjectSchemaExtensions
+{
+    private static readonly Dictionary<Type, OpenApiSchema> _simpleTypeToOpenApiSchema = new()
+    {
+        [typeof(bool)] = new() { Type = "boolean" },
+        [typeof(byte)] = new() { Type = "string", Format = "byte" },
+        [typeof(int)] = new() { Type = "integer", Format = "int32" },
+        [typeof(uint)] = new() { Type = "integer", Format = "int32" },
```

Comment:
This seems a little strange.  If the types don't need to match exactly, why not also use `int32` for `short` and `ushort`?

Reply:
I'm not sure what you mean by "if the types don't need to match exactly" but the reason there is no format set for `short` and `ushort` is because [the spec doesn't define one](https://spec.openapis.org/oas/v3.0.3#data-types).

Reply:
Won't the int32 checks rule out valid uints?